include/envoy/upstream/cluster_manager.h

```diff
@@ -174,11 +174,12 @@ class ClusterManager {
     }
 
     ClusterConstOptRef getCluster(absl::string_view cluster) {
-      auto active_cluster = active_clusters_.find(cluster);
+      auto cluster_name = absl::AsciiStrToLower(cluster);
```

Comment:
I think this can technically be considered a breaking API change, to @ggreenway point this would conflate two valid distinct clusters `FooX` and `foox`.  Which situation do we expect cluster names to be DNS names that need normalizing?

This is tangentially related to recent discussion with @envoyproxy/api-shepherds on how to normalize names for lookup with `xdstp://`, we are planning on a more extensive `ClusterHandle` approach.

I would remove this normalization unless there is a good rationale.